src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  addQueryID,
+  debounce,
+  addAbsolutePosition,
+  noop,
+  escapeHits,
+} from '../../lib/utils';
+import {
+  Connector,
+  Hits,
+  Hit,
+  FindAnswersOptions,
+  FindAnswersResponse,
+} from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'answers',
+  connector: true,
+});
+
+export type AnswersRendererOptions = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Hits;
+
+  /**
+   * whether it's still loading the results from the /answers API
+   */
+  isLoading: boolean;
+};
+
+export type AnswersConnectorParams = {
+  /**
+   * Attributes to use for predictions.
+   * If empty, we use all `searchableAttributes` to find answers.
+   * All your `attributesForPrediction` must be part of your `searchableAttributes`.
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: Array<'en'>;
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce render
+   * @default 100
+   */
+  renderDebounceTime?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce search
+   * @default 100
+   */
+  searchDebounceTime?: number;
+
+  /**
+   * Whether to escape HTML tags from hits string values.
+   *
+   * @default true
+   */
+  escapeHTML?: boolean;
+
+  /**
+   * Extra parameters to pass to findAnswers method.
+   * @default {}
+   */
+  extraParameters?: FindAnswersOptions;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const {
+      queryLanguages,
+      attributesForPrediction,
+      nbHits = 1,
+      renderDebounceTime = 100,
+      searchDebounceTime = 100,
+      escapeHTML = true,
+      extraParameters = {},
+    } = widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
```

Comment:
Do we want to default `queryLanguages` to `['en']`? (Especially because this is the only supported value.)

Reply:
it was an explicit choice from the answers team to make users pass queryLanguages, so later when other languages are added, the language isn't wrong